### PR TITLE
getStatistic method done

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,55 @@
...
+            }
+            resultFileMap.put("result", resultFileMap.get("supply") - resultFileMap.get("buy"));
+
+            for (Map.Entry<String, Integer> entry : resultFileMap.entrySet()) {
+                String lineToWrite = entry.getKey() + "," + entry.getValue();
+                writeTo.write(lineToWrite);
+                writeTo.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cant open the file.", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:51]: add file name

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromFile = new File(fromFileName);
+        File toFile = new File(toFileName);
+        Map<String, Integer> resultFileMap = new LinkedHashMap<>();
+        resultFileMap.put("supply", ZERO_NUMBER);
+        resultFileMap.put("buy", ZERO_NUMBER);
+        resultFileMap.put("result", ZERO_NUMBER);
+
+        try (BufferedReader readerFrom = new BufferedReader(new FileReader(fromFile));
+                BufferedWriter writeTo = new BufferedWriter(new FileWriter(toFile))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:26]: please create separate private helper methods for reading and writing and move logic of reading and writing to them

you will call these methods inside getStatistic()

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,67 @@
...
+    public static final int ZERO_NUMBER = 0;
+    public static final int MAP_INDEX_KEY = 0;
+    public static final int MAP_INDEX_VALUE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> resultFileMap = readFile(fromFileName);
+        writeFile(toFileName, resultFileMap);
+    }
+
+    public Map<String, Integer> readFile(String fileName) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:22]: please make helper methods private

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,67 @@
...
+        try (BufferedReader readerFrom = new BufferedReader(new FileReader(fromFile))) {
+            String value = readerFrom.readLine();
+
+            while (value != null) {
+                String[] valueStringArray = value.split(",");
+
+                if (fileMap.containsKey(valueStringArray[MAP_INDEX_KEY])) {
+                    int sumNumbers = fileMap.get(valueStringArray[MAP_INDEX_KEY]);
+                    fileMap.put(valueStringArray[MAP_INDEX_KEY],
+                            (sumNumbers + Integer.parseInt(valueStringArray[1])));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:38]:
```suggestion
                            (sumNumbers + Integer.parseInt(valueStringArray[MAP_INDEX_VALUE])));
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,67 @@
...
+        fileMap.put("buy", ZERO_NUMBER);
+        fileMap.put("result", ZERO_NUMBER);
+
+        try (BufferedReader readerFrom = new BufferedReader(new FileReader(fromFile))) {
+            String value = readerFrom.readLine();
+
+            while (value != null) {
+                String[] valueStringArray = value.split(",");
+
+                if (fileMap.containsKey(valueStringArray[MAP_INDEX_KEY])) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:35]: put valueStringArray[MAP_INDEX_KEY] and valueStringArray[MAP_INDEX_VALUE] to local variables and reuse them

your code will look cleaner

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,67 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException("Cant open the file " + fileName, e);
+        }
+        return fileMap;
+    }
+
+    public void writeFile(String fileName, Map<String, Integer> fileMap) {
+        File toFile = new File(fileName);
 
+        try (BufferedWriter writeTo = new BufferedWriter(new FileWriter(toFile))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:57]:
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFile))) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,67 @@
...
+    }
+
+    public Map<String, Integer> readFile(String fileName) {
+        File fromFile = new File(fileName);
+        Map<String, Integer> fileMap = new LinkedHashMap<>();
+        fileMap.put("supply", ZERO_NUMBER);
+        fileMap.put("buy", ZERO_NUMBER);
+        fileMap.put("result", ZERO_NUMBER);
+
+        try (BufferedReader readerFrom = new BufferedReader(new FileReader(fromFile))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:29]:
```suggestion
        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,7 +1,67 @@
...
+    public static final int MAP_INDEX_VALUE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> resultFileMap = readFile(fromFileName);
+        writeFile(toFileName, resultFileMap);
+    }
+
+    public Map<String, Integer> readFile(String fileName) {
+        File fromFile = new File(fileName);
+        Map<String, Integer> fileMap = new LinkedHashMap<>();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]:
```suggestion
        Map<String, Integer> resultMap = new LinkedHashMap<>();
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 5 above)
Comment (Vasul-Pankiv) [reply src/main/java/core/basesyntax/WorkWithFile.java]: Not resolved

**Review by nacenik — APPROVED** [review]
LGTM! 👍  But don't use advanced features in the future. All tasks that you can solve with knowledge of the current topic

**Pull request closed without merging**